src/Microsoft.PowerShell.Commands.Utility/commands/utility/join-string.cs

```diff
@@ -0,0 +1,210 @@
+            set => _separator = value;
+        }
+
+        /// <summary>
+        /// Gets or sets text to include before the joined input text.
+        /// </summary>
+        [Parameter]
+        [Alias("op")]
+        public string OutputPrefix { get; set; }
+
+        /// <summary>
+        /// Gets or sets text to include after the joined input text.
+        /// </summary>
+        [Parameter]
+        [Alias("os")]
+        public string OutputSuffix { get; set; }
+
+        /// <summary>
+        /// Gets or sets if the output items should we wrapped in single quotes.
+        /// </summary>
+        [Parameter(ParameterSetName = "SingleQuote")]
+        public SwitchParameter SingleQuote { get; set; }
+
+        /// <summary>
+        /// Gets or sets if the output items should we wrapped in double quotes.
+        /// </summary>
+        [Parameter(ParameterSetName = "DoubleQuote")]
+        public SwitchParameter DoubleQuote { get; set; }
+
+        /// <summary>
+        /// Gets or sets a format string that is applied to each input object.
+        /// </summary>
+        [Parameter(ParameterSetName = "Format")]
+        [ArgumentCompleter(typeof(JoinItemCompleter))]
+        public string FormatString { get; set; }
+
+        /// <summary>
+        /// Gets or sets the input object to join into text.
+        /// </summary>
+        [Parameter(ValueFromPipeline = true)]
+        public PSObject InputObject { get; set; }
+
+        /// <inheritdoc />
+        protected override void BeginProcessing()
+        {
+            _quoteChar = SingleQuote ? '\'' : DoubleQuote ? '"' : char.MinValue;
+            _outputBuilder.Append(OutputPrefix);
+        }
+
+        /// <inheritdoc />
+        protected override void ProcessRecord()
+        {
+            if (InputObject != null && InputObject != AutomationNull.Value)
+            {
+                var inputValue = Property == null
+                                    ? InputObject
+                                    : Property.GetValues(InputObject, false, true).FirstOrDefault()?.Result;
+                var stringValue = LanguagePrimitives.ConvertTo<string>(inputValue);
+
+                if (_firstInputObject)
+                {
+                    _firstInputObject = false;
+                }
+                else
+                {
+                    _outputBuilder.Append(Separator);
+                }
+
+                if (_quoteChar != char.MinValue)
+                {
+                    _outputBuilder.Append(_quoteChar);
+                    _outputBuilder.Append(stringValue);
+                    _outputBuilder.Append(_quoteChar);
+                }
+                else if (string.IsNullOrEmpty(FormatString))
+                {
+                    _outputBuilder.Append(stringValue);
+                }
+                else
+                {
+                    _outputBuilder.AppendFormat(CultureInfo.CurrentCulture, FormatString, stringValue);
```

Comment:
I'd want to get confirmation that this should be `CurrentCulture` not `InvariantCulture`.
And maybe discuss `-FormatCulture` parameter.

Reply:
Same here. Now sure what is right here.

Reply:
PowerShell consistently uses the invariant culture in to/from-string conversions, so it should probably be `InvariantCulture`, at least by default, perhaps with an optional  `-UseCulture` switch (analogous to the `*-Csv` cmdlets).

Reply:
```powershell
 gcm -ParameterName Culture

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Cmdlet          Compare-Object                                     3.1.0.0    Microsoft.PowerShell.Utility
Cmdlet          Group-Object                                       3.1.0.0    Microsoft.PowerShell.Utility
Cmdlet          New-PSSessionOption                                6.0.0.0    Microsoft.PowerShell.Core
Cmdlet          Sort-Object                                        3.1.0.0    Microsoft.PowerShell.Utility
```

We already have `-Culture` parameter. Make sense re-use the name?

Reply:
@iSazonov: 

I think `-UseCulture` is the correct name, after all:

* The `-Culture <string>` parameters you reference:
    * require an _argument_, namely a _given_ culture
    * are used with cmdlets that are current-culture-_sensitive_ by default, to allow opt-in to a _different_ culture.

* The `-UseCulture` switch, by contrast:
    * is used with cmdlets that are culture-_invariant_ by default, to allow opt-in to the _current_ culture (currently applies only to the `*-Csv` cmdlets).

The latter is what is being implemented here.

Reply:
@mklement0 Thanks! We need discuss Select-String  too.